src/README.md

```diff
@@ -1,17 +1,22 @@
-Welcome to the source of ReasonReact!
+### Welcome to the source of ReasonReact
+
+We want to expose as minimum modules to the toplevel as possible, and at the same time want to map closely to the npm packages. So each module maps to a npm package, and each sub-module maps to a sub-module. For example: `react-dom` -> `ReactDOM` and `react-dom/server` -> `ReactDOM.Server`.
 
 Files overview:
 
 ## Bindings
 
-- `React`: bindings to React.
-- `ReactDOM`: bindings to ReactDOM.
-- `ReactDOMServer`: bindings to ReactDOMServer.
-- `ReactEvent`: bindings to React's custom events system.
-- `ReactDOMStyle`: bindings to create `style` objects.
+- `React`: bindings to React
+  |- `React.Event`: bindings to React's custom events system
+  |- `React.Context`: bindings to React's Context
+- `ReactDOM`: bindings to ReactDOM
+  |- `ReactDOM.Server`: bindings to ReactDOMServer
+  |- `ReactDOM.Style`: bindings to create `style` objects
+  |- `ReactDOM.TestUtils`: helpers for testing your components
 
-## Extra
+## Extra (not part of React)
 
-- `ReactTestUtils`: helpers for testing your components
-- `ReasonReactErrorBoundary`: component to catch errors within your component tree
 - `ReasonReactRouter`: a simple, yet fully featured router with minimal memory allocations
+- `ErrorBoundary`: component to catch errors within your component tree
```

Comment:
should this one become `React.ErrorBoundary`, or do you want it to stay out at the toplevel?

Reply:
OK then I'm fine taking the time to think about this, and keeping it `ReasonReactErrorBoundary` for now?

Reply:
Sounds perfect